posydon/popsyn/binarypopulation.py

```diff
@@ -343,12 +343,15 @@ def _safe_evolve(self, **kwargs):
                     e.add_note(initial_condition_message(binary))
                     traceback.print_exception(e)
 
-            binary.warnings = []
-            if len(w) > 0:
-                for x in w: 
-                    warning_formatted = warnings.formatwarning(x.message, x.category, x.filename, x.lineno, line=None)
-                    print(warning_formatted, file=sys.stderr)
-                    binary.warnings.append(warning_formatted)
+                ## catching warnings with record=True automatically suppresses them being written standard error,
+                ## but we need record=True to save all warnings as a binary attribute (binary.warnings)
+                ## so, we manually reformat them and print to stderr
+                binary.warnings = []
+                if len(w) > 0:
+                    for x in w: 
+                        warning_formatted = warnings.formatwarning(x.message, x.category, x.filename, x.lineno, line=None)
+                        print(warning_formatted, file=sys.stderr)
```

Comment:
I'm not sure, why you don't use the `warn` function here. With simply printing the formatted warning you build a workaround the filtering of warnings, which I think isn't your aim here.

Reply:
The filtering and printing are two different things. We want to catch and record all of the warnings, and the warnings that are recorded depends on the filters. We need to record the warnings if we want to keep the functionality of saving a binary's warnings as an attribute. However, recording the warnings suppresses them being printed to output, so they must be manually printed.

Reply:
So if you filter-out the POSYDON warnings, they are not included in "w". If you want to keep the POSYDON warnings, they are saved in "w".

Reply:
OK. I already had in mind the replacement I was building for the catch. So that we can get other functionality on top of it. Additionally, I was not that happy about the catching function, because it is classified as not being thread save.